tyro/_cli.py

```diff
@@ -241,18 +293,26 @@ def _cli_impl(
 
     # Read and fix arguments. If the user passes in --field_name instead of
     # --field-name, correct for them.
-    args = sys.argv[1:] if args is None else args
-
-    def fix_arg(arg: str) -> str:
-        if not arg.startswith("--"):
-            return arg
-        if "=" in arg:
-            arg, _, val = arg.partition("=")
-            return arg.replace("_", "-") + "=" + val
-        else:
-            return arg.replace("_", "-")
-
-    args = list(map(fix_arg, args))
+    args = list(sys.argv[1:]) if args is None else list(args)
+
+    # Fix arguments. This will modify all option-style arguments replacing
+    # underscores with dashes. This is to support the common convention of using
+    # underscores in variable names, but dashes in command line arguments.
+    # If two options are ambiguous, e.g., --a_b and --a-b, raise a runtime error.
+    modified_args: Dict[str, str] = {}
+    for index, arg in enumerate(args):
+        if arg.startswith("--"):
```

Comment:
(no fix needed, unless you feel like it)
For this kind of condition I personally prefer:
```python
if not arg.startswith("--"):
    continue
```
which adds 1 line, but I find more readable because it reduces the nesting